### PR TITLE
Move graph validation errors to the entities causing the errors

### PR DESCRIPTION
**Pull request by jacobwindsor**
dcp-507

- Removes `graphValidatonErrors` from `SubmissionEnvelope`
- Adds `graphValidationErrors` to `MetadataDocument` and makes it patchable in biomaterial, process, protocol, file
- Adds migration to add `graphValidationErrors` as an empty array to biomaterial, process, protocol, file
- Adds `graphInvalid[entity]` to submission envelope summary
- Removes graphValidationErrors from all entities in a submission when a submission goes from INVALID -> another state
- Adds tests

### CODE REVIEW

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelope.java [hunk 1]
```diff
@@ -30,8 +30,6 @@
     private SubmissionState submissionState;
     private SubmissionGraphValidationState graphValidationState;
     private @Setter
-    List<GraphValidationError> graphValidationErrors;
```
Comment (aaclan-ebi) [inline src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelope.java:33]: why remove this?

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java [hunk 2]
```diff
@@ -291,4 +295,30 @@ private boolean isSubmitAction(List<SubmitAction> submitActions) {
...
+                    protocolRepository.save(protocol);
+                });
+
+        fileRepository.findBySubmissionEnvelope(submissionEnvelope)
+                .forEach(file -> {
+                    file.setGraphValidationErrors(new ArrayList<>());
+                    fileRepository.save(file);
+                });
+    }
 }
```
Comment (aaclan-ebi) [inline src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java:324]: could optimise by doing an update with condition

**Review by aaclan-ebi — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelope.java (hunk 1 above)
Comment (jacobwindsor) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelope.java:33]: Errors are now on the entities that caused them

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java (hunk 2 above)
Comment (jacobwindsor) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java:324]: Can you explain a bit more?

Comment (aaclan-ebi) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java:324]: I thought you can do update many using the mongo repository object. Looks like it's not supported. https://github.com/spring-projects/spring-data-mongodb/blob/e28bede416e4ddac19a35dc239388afc90b9cac4/src/main/asciidoc/reference/mongo-repositories.adoc

You could also use saveAll. There's a slight performance improvement. Found this:
https://www.baeldung.com/spring-data-save-saveall

**Review by ke4 — APPROVED** [review]

File: src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java (hunk 2 above)
Comment (jacobwindsor) [reply src/main/java/org/humancellatlas/ingest/submission/SubmissionEnvelopeService.java]: Okay I used saveAll

**Review by ke4 — APPROVED** [review]
LGTM 👍

**Review by aaclan-ebi — APPROVED** [review]

**Pull request merged**